spec/lib/declare_schema/migration_generator_spec.rb

```diff
@@ -585,21 +586,64 @@ class Advert < ActiveRecord::Base # rubocop:disable Lint/ConstantDefinitionInBlo
 
       Advert.index_definitions.delete_if { |spec| spec.fields == ["title"] }
 
+      ActiveRecord::Migration.class_eval(up)
+
+      # You can migrate an index to have slightly different settings
+
+      class Advert < ActiveRecord::Base # rubocop:disable Lint/ConstantDefinitionInBlock
+        declare_schema do
+          string :title, limit: 250, null: true, index: { unique: true, name: 'my_index', length: 5 }
+        end
+      end
+
+      # Note: the index is removed first, then re-added so that we don't get an error about the index already existing.
+      expect(Generators::DeclareSchema::Migration::Migrator.run).to(
+        migrate_up(<<~EOS.strip)
+          remove_index :adverts, name: :my_index
+          add_index :adverts, [:title], name: :my_index, unique: true, length: { title: 5 }
+        EOS
+      )
+
+      ActiveRecord::Migration.class_eval(down)
+
       # You can create an index on more than one field
 
       class Advert < ActiveRecord::Base # rubocop:disable Lint/ConstantDefinitionInBlock
+        declare_schema do
+          string :title, limit: 250, null: true, index: { unique: true, name: 'my_index', length: 5 }
+          integer :category_id, limit: 8
+        end
         index [:title, :category_id]
       end
 
-      expect(Generators::DeclareSchema::Migration::Migrator.run).to(
+      up, down = Generators::DeclareSchema::Migration::Migrator.run
+      expect([up, down]).to(
         migrate_up(<<~EOS.strip)
           add_column :adverts, :title, :string, limit: 250, null: true#{charset_and_collation}
+          add_column :adverts, :category_id, :integer, limit: 8, null: false
+          add_index :adverts, [:title], name: :my_index, unique: true, length: { title: 5 }
           add_index :adverts, [:title, :category_id], name: :index_adverts_on_title_and_category_id
         EOS
       )
 
+      ActiveRecord::Migration.class_eval(up)
+
       Advert.index_definitions.delete_if { |spec| spec.fields == ["title", "category_id"] }
 
+      # You can change the column order of an index
```

Comment:
This is the spec for the secondary bug. Just swapping the order of the columns now causes the index to be dropped then added below.